pkg/cri/sbserver/podsandbox/controller.go

```diff
@@ -38,11 +40,38 @@ import (
 	"github.com/containerd/containerd/sandbox"
 )
 
-// CRIService interface contains things required by controller, but not yet refactored from criService.
-// TODO: this will be removed in subsequent iterations.
-type CRIService interface {
-	// TODO: we should implement Event backoff in Controller.
-	BackOffEvent(id string, event interface{})
+func init() {
+	plugin.Register(&plugin.Registration{
+		Type: plugin.GRPCPlugin,
+		ID:   "cri-podsandbox-controller",
+		Requires: []plugin.Type{
+			plugin.InternalPlugin,
+		},
+		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
+			// Get base CRI dependencies.
+			criBasePlugin, err := ic.GetByID(plugin.InternalPlugin, "cri-base")
+			if err != nil {
+				return nil, fmt.Errorf("unable to load CRI service base dependencies: %w", err)
+			}
+			criBase := criBasePlugin.(*base.CRIBase)
+
+			// Get image service.
+			criImagePlugin, err := ic.GetByID(plugin.GRPCPlugin, "cri-image-service")
```

Comment:
This is not a safe dependency. I'm trying to carve out the client dependency and image service. If we define the interface needed here, we can probably break this out cleaner.

Reply:
*Briefly synced offline*
I don't think there is a clean way to define these dependencies without reworking `plugin/` package (or without breaking changes in configuration). Do you want to postpone this PR until we figure something out or its fine to leave a TODO for follow up work and move on with further decoupling?